Tests/MongoSwiftSyncTests/SpecTestRunner/TestOperation.swift

```diff
@@ -599,122 +847,432 @@ struct RenameCollection: TestOperation {
+        return nil
+    }
+}
+
+struct CreateIndex: TestOperation {
+    let session: String
+    let name: String
+    let keys: Document
+
+    private enum CodingKeys: String, CodingKey { case session, name, keys }
+
+    init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        self.session = (try? container.decode(String.self, forKey: .session)) ?? ""
+        self.name = try container.decode(String.self, forKey: .name)
+        self.keys = try container.decode(Document.self, forKey: .keys)
+    }
+
+    func execute(on target: TestOperationTarget, sessionDict: [String: ClientSession]) throws -> TestOperationResult? {
+        guard case let .collection(collection) = target else {
+            throw TestError(message: "collection not provided to createIndex")
+        }
+        let indexOptions = IndexOptions(name: self.name)
+        _ = try collection.createIndex(self.keys, indexOptions: indexOptions, session: sessionDict[self.session])
+        return nil
+    }
+}
+
+struct RunCommand: TestOperation {
+    let session: String
+    let command: Document
+    let readPreference: ReadPreference
+
+    private enum CodingKeys: String, CodingKey { case session, command, readPreference }
+
+    init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        self.session = (try? container.decode(String.self, forKey: .session)) ?? ""
+        self.command = try container.decode(Document.self, forKey: .command)
+        self.readPreference = (try? container.decode(ReadPreference.self, forKey: .readPreference)) ??
+            ReadPreference.primary
+    }
+
+    func execute(on target: TestOperationTarget, sessionDict: [String: ClientSession]) throws -> TestOperationResult? {
+        guard case let .database(database) = target else {
+            throw TestError(message: "database not provided to runCommand")
+        }
+        let runCommandOptions = RunCommandOptions(readPreference: self.readPreference)
+        let result = try database.runCommand(
+            self.command,
+            options: runCommandOptions,
+            session: sessionDict[self.session]
+        )
+
+        var refinedResult = Document()
+        try result.copyElements(
+            to: &refinedResult,
+            excluding: [
+                "opTime", "electionId", "ok", "$clusterTime", "signature", "keyId", "operationTime",
+                "recoveryToken"
+            ]
+        )
+        return TestOperationResult(from: refinedResult)
+    }
+}
+
+struct AssertCollectionExists: TestOperation {
+    let database: String
+    let collection: String
+
+    func execute(on target: TestOperationTarget, sessionDict _: [String: ClientSession])
+        throws -> TestOperationResult? {
+        guard case let .testRunner(database) = target else {
+            throw TestError(message: "database not provided to assertCollectionExists")
```

Comment:
rather than throwing errors here and in the other `.testRunner` operations, I think we should instead be doing actual assertions (i.e. `expect(...)`)

Reply:
Ok, I've added the expectations. Note that we still need the `TestError`s in the first guard statement since we expect the test runner to contain the `MongoDatabase`.